sdk/src/azure/core/az_http_policy_retry.c

```diff
@@ -220,6 +220,4 @@ AZ_NODISCARD az_result az_http_pipeline_policy_retry(
       }
     }
   }
-
-  return result;
```

Comment:
How is it guaranteed that we won't be stuck in an infinite loop?

I think a comment above this line would be helpful for maintainability in the future:
> if (attempt > max_retries || az_result_failed(result))

Something like:
// We increment attempt, each iteration of the loop, and this condition guarantees the loop will terminate after hitting max retries.

Reply:
There are other conditions that would terminate the loop.  Letting the code self-document seems better in this case, less likely the comment gets out of sync.
![image](https://user-images.githubusercontent.com/4430337/174126280-1144423b-1df8-486f-b12b-9c154d6ea820.png)